Checking experiments-msmarco-v2.md and reproducing results. (#1611)
castorini/anserini: docs/experiments-msmarco-v2.md

This guide presents information for working with V2 of the MS MARCO passage and document test collections, available [here](https://microsoft.github.io/msmarco/TREC-Deep-Learning.html).

If you're having issues downloading the collection via `wget`, try using [AzCopy](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-v10).
For example, to download passage collection,
```
azcopy copy https://msmarco.blob.core.windows.net/msmarcoranking/msmarco_v2_passage.tar ./collections
```

> The speedup using `azcopy` here is significant, in our case, it takes ~1 to 2min to download this tarball, compared to 2+ hours when using `wget`.

Reviewer: ```suggestion
The speedup using `azcopy` is significant compared to `wget`, but the actual downloading time will vary based on your location as well as many other factors.
```

The complete index occupies 72 GB (138,364,198 passages).
It's big because it includes postions (for phrase queries), document vectors (for relevance feedback), and a complete copy of the collection itself.
The index size as well as index time can be reduced by removing the options `-storePositions`, `-storeDocvectors`, `-storeRaw` as appropriate.
For reference:

> + Without any of the three above option, index size reduces to 12 GB (indexed in ~13min).

Reviewer: ```suggestion
+ Without any of the three above option, index size reduces to 12 GB.
```

> + With just `-storeRaw`, index size reduces to 47 GB (indexed in ~17min). This setting contains the raw JSON document, which makes it suitable for use as first-stage retrieval to support downstream rerankers. Bloat compared to compressed size of raw collection is due to support for per-document random access.

Reviewer: ```suggestion
+ With just `-storeRaw`, index size reduces to 47 GB. This setting contains the raw JSON document, which makes it suitable for use as first-stage retrieval to support downstream rerankers. Bloat compared to compressed size of raw collection is due to support for per-document random access.
```

There are a total of 138,364,198 passages in the collection (exactly the same as the original passage collection).
In each "document" in the index comprises the url, title, headings, and passage fields concatenated together.

> With the above indexing configuration, the index size comes to 162 GB (indexed in ~40min).

Reviewer: ```suggestion
With the above indexing configuration, the index size comes to 162 GB.
```

However, the index size can be reduced by playing with the indexing options discussed above.

> For example, with just the `-storeRaw` option, which supports bag-of-words first-stage retrieval with stored raw documents that can be fetched and passed to a downstream reranker, the index size comes out to 95 GB (~28 min index time).

Reviewer: ```suggestion
For example, with just the `-storeRaw` option, which supports bag-of-words first-stage retrieval with stored raw documents that can be fetched and passed to a downstream reranker, the index size comes out to 95 GB.
```

```

Same instructions as above.

> On the same machine, indexing takes around 34min.

Reviewer: ```suggestion
On the same machine, indexing takes around 40 minutes.
```

Complete index occupies 134 GB (11,959,635 documents).
Index size can be reduced by removing the options `-storePositions`, `-storeDocvectors`, `-storeRaw` as appropriate.
For reference:

> + Without any of the three above option, index size reduces to 9.4 GB (indexed in ~18min).

Reviewer: ```suggestion
+ Without any of the three above option, index size reduces to 9.4 GB.
```

> + With just `-storeRaw`, index size reduces to 73 GB (indexed in ~25min). This setting contains the raw JSON document, which makes it suitable for use as first-stage retrieval to support downstream rerankers. Bloat compared to compressed size of raw collection is due to support for per-document random access; evidently, the JSON docs don't compress well.

Reviewer: ```suggestion
+ With just `-storeRaw`, index size reduces to 73 GB. This setting contains the raw JSON document, which makes it suitable for use as first-stage retrieval to support downstream rerankers. Bloat compared to compressed size of raw collection is due to support for per-document random access; evidently, the JSON docs don't compress well.
```

There are a total of 124,131,414 "documents" in the collection.
Each "document" comprises the url, title, headings, and segment fields concatenated together.

> With the above indexing configuration, the index size comes to 226 GB (indexed in ~1h).

Reviewer: ```suggestion
With the above indexing configuration, the index size comes to 226 GB.
```

However, the index size can be reduced by playing with the indexing options discussed above.

> For example, with just the `-storeRaw` option, which supports bag-of-words first-stage retrieval with stored raw documents that can be fetched and passed to a downstream reranker, the index size comes out to 124 GB (indexed in ~48min).

Reviewer: ```suggestion
For example, with just the `-storeRaw` option, which supports bag-of-words first-stage retrieval with stored raw documents that can be fetched and passed to a downstream reranker, the index size comes out to 124 GB.
```
